### PR TITLE
Cov online & batch & distr missed checks

### PR DESCRIPTION
**Pull request by mchernov-intel**

### CODE REVIEW

**Comment by mchernov-intel** [comment]
./intelci: run

**Comment by mchernov-intel** [comment]
@vepifanov debug build checked locally

File: cpp/daal/src/algorithms/covariance/oneapi/covariance_dense_distr_step2_oneapi_impl.i [hunk 1]
```diff
@@ -50,8 +50,13 @@ services::Status CovarianceDenseDistrStep2KernelOneAPI<algorithmFPType, method>:
 
     services::Status status;
 
+    if (crossProductTable->getNumberOfColumns() > static_cast<size_t>(services::internal::MaxVal<uint32_t>::get()))
+    {
+        return services::Status(daal::services::ErrorCovarianceInternal);
+    }
+
+    const uint32_t nFeatures    = static_cast<uint32_t>(crossProductTable->getNumberOfColumns());
```
Comment (amgrigoriev) [inline cpp/daal/src/algorithms/covariance/oneapi/covariance_dense_distr_step2_oneapi_impl.i:58]: It looks like number of rows should be equal to a number of columns for the table. But there is no check for that.

File: cpp/daal/src/algorithms/covariance/oneapi/covariance_oneapi_impl.i [hunk 2]
```diff
@@ -180,33 +185,34 @@ services::Status updateDenseCrossProductAndSums(bool isNormalized, size_t nFeatu
...
-
     services::Status status;
 
     auto & context = services::internal::getDefaultContext();
     auto & factory = context.getClKernelFactory();
-    __buildProgram<algorithmFPType>(factory);
+    status |= __buildProgram<algorithmFPType>(factory);
+    DAAL_CHECK_STATUS_VAR(status);
 
     auto kernel = factory.getKernel("mergeCrossProduct");
```
Comment (amgrigoriev) [inline cpp/daal/src/algorithms/covariance/oneapi/covariance_oneapi_impl.i:201]: getKernel - check status

Comment (mchernov-intel) [reply cpp/daal/src/algorithms/covariance/oneapi/covariance_oneapi_impl.i]: fixed

File: cpp/daal/src/algorithms/covariance/oneapi/covariance_dense_distr_step2_oneapi_impl.i (hunk 1 above)
Comment (mchernov-intel) [reply cpp/daal/src/algorithms/covariance/oneapi/covariance_dense_distr_step2_oneapi_impl.i:58]: Size checks implemented via `DAAL_ASSERT_UNIVERSAL_BUFFER` are everywhere before each kernel call. So, the is no reason to put additional check here.

**Comment by mchernov-intel** [comment]
@amgrigoriev @vepifanov please, review

**Review by vepifanov — APPROVED** [review]
LGTM.

**Pull request merged**